lib/setupHooks/installCargoArtifactsHook.sh

```diff
@@ -1,20 +1,46 @@
 compressAndInstallCargoArtifactsDir() {
   local dir="${1:?destination directory not defined}"
   local cargoTargetDir="${2:?cargoTargetDir not defined}"
+  local prevArtifacts="${3}"
 
   mkdir -p "${dir}"
 
   local dest="${dir}/target.tar.zst"
   echo "compressing ${cargoTargetDir} to ${dest}"
   (
     export SOURCE_DATE_EPOCH=1
-    tar --sort=name \
+
+    dynTar() {
+      if [ -n "${doCompressAndInstallFullArchive}" ]; then
+        >&2 echo "compressing and installing full archive as requested"
+        tar "$@" "${cargoTargetDir}"
+      elif [ "$(uname -s)" == "Darwin" ]; then
+        # https://github.com/rust-lang/rust/issues/115982
+        >&2 echo "incremental zstd compression not currently supported on Darwin: https://github.com/rust-lang/rust/issues/115982"
+        tar "$@" "${cargoTargetDir}"
+      elif [ -z "${prevArtifacts}" ]; then
+        >&2 echo "no previous artifacts found, compressing and installing full archive"
+        tar "$@" "${cargoTargetDir}"
+      else
+        >&2 echo "linking previous artifacts ${prevArtifacts}"
```

Comment:
The output here is confusing and might get people to misread the times how long thing took.

`echo "compressing ${cargoTargetDir} to ${dest}"` should happen right before the actual `tar | zstd`, IMO.

Reply:
I can tweak the logs here